api/src/pcapi/utils/login_manager.py

```diff
@@ -24,19 +24,34 @@ def get_request_authorization() -> werkzeug.datastructures.Authorization | None:
 
 
 @app.login_manager.user_loader  # type: ignore [attr-defined]
-def get_user_with_id(user_id: int) -> users_models.User | None:
+def get_user_with_id(user_id: str) -> users_models.User | None:
     flask.session.permanent = True
     session_uuid = flask.session.get("session_uuid")
     if not users_models.UserSession.query.filter_by(userId=user_id, uuid=session_uuid).one_or_none():
         return None
 
     try:
         if flask.request.blueprint.startswith(backoffice_web.name):  # type: ignore [union-attr]
-            return backoffice_api.fetch_user_with_profile(user_id)
+            return backoffice_api.fetch_user_with_profile(int(user_id))
     except AttributeError:
         pass
 
-    return users_models.User.query.filter(users_models.User.id == user_id).one_or_none()
+    internal_admin_id = flask.session.get("internal_admin_id", 0)
+    internal_admin = None
+    user = None
+    users = users_models.User.query.filter(
+        users_models.User.id.in_([user_id, internal_admin_id]),
+    ).all()
+    for candidate in users:
+        if candidate.id == int(user_id):
+            user = candidate
+        elif candidate.id == internal_admin_id:
+            internal_admin = candidate
```

Comment:
Peut-être une question de goût, je n'aime pas vraiment ce code avec une boucle pour départager alors qu'on veut juste charger un tuple. Surtout avec `in([user_id, None])` suivi d'une boucle, dans la grande majorité des requêtes.
Pourquoi pas plutôt une requête appelant deux select en sous-requête, donc renvoyant un tuple d'ordre déterminé, dans le cas où on a un `internal_admin_id`, et garder le fonctionnement actuel dans le cas nominal ?

Reply:
La solution envisagée ne marche malheueusement pas donc on a décider de passer sur é requetes séparés. les perfs ne devraient pas être trop impactés (le cas reste rare et les requetes sont légeres) et le code sera beaucoup plus lisible.